src/cursor/cur_std.c

```diff
@@ -1052,6 +1052,15 @@ __wt_cursor_reconfigure(WT_CURSOR *cursor, const char *config)
     } else
         WT_ERR_NOTFOUND_OK(ret, false);
 
+    /* Set the prefix search near flag. */
```

Comment:
can we move this (indicating key set is prefix) bit to set_key, by doing so I think we can control what type of cursors could support prefix search?
I am assuming this is possible as set_key takes variable arguements.